keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c

```diff
@@ -0,0 +1,256 @@
+                    isMacMode = false;
+                }
+            }
+            return false;
+
+        case KC_PC1:
+            if (record->event.pressed) {
+                if (kvm_pc_sel) {
+                    kvm_pc_sel = 0;
+                    kvm_switch(0);
+
+                    keyboard_config.eeprom_kvm_pc_sel = kvm_pc_sel;
+                    eeconfig_update_kb(keyboard_config.raw); // Writes the new status to EEPROM
+                }
+            }
+            return false;
+
+        case KC_PC2:
+            if (record->event.pressed) {
+                if (!kvm_pc_sel){
+                    kvm_pc_sel = 1;
+                    kvm_switch(1);
+
+                    keyboard_config.eeprom_kvm_pc_sel = kvm_pc_sel;
+                    eeconfig_update_kb(keyboard_config.raw); // Writes the new status to EEPROM
+                }
+            }
+            return false;
+
+        case KC_TPC:
+            if (record->event.pressed) {
+                kvm_pc_sel = !kvm_pc_sel;
+                if (kvm_pc_sel) kvm_switch(1);
+                else kvm_switch(0);
+                keyboard_config.eeprom_kvm_pc_sel = kvm_pc_sel; //
+                eeconfig_update_kb(keyboard_config.raw); // Writes the new status to EEPROM
+            }
+            return false;
+
+        case KC_S_LED:                                  // Change Status LED Brightness
+            if (record->event.pressed) {
+                if (++Status_LED_Bright > 5) {
+                    Status_LED_Bright = 1;
+                };
+                keyboard_config.eeprom_Status_LED_Bright = Status_LED_Bright; //
+                eeconfig_update_kb(keyboard_config.raw); // Writes the new status to EEPROM
+            }
+            return false;
+
+        default:
+            return true;   // Process all other keycodes normally
+    }
+    return true;
+}
+
+bool isRecording = false;           // dynamic macro REC. LED
+bool isRecordingLedOn = false;
+static uint16_t recording_timer;
+
+void housekeeping_task_kb(void) {
+
+    if (kvm_sel_on == true){
+        if (timer_elapsed(kvm_timer) > kvm_deadtime ) {
+            gpio_write_pin_high(GPIO_KM_PWEN);
+            kvm_sel_on = false;                             //// 키보드 리셋이 필요?
+        }
+    }
+
+    if (INIT3S_on == true){
+        if (timer_elapsed(INIT3S_timer) > 3000){
+            INIT3S_on = false;
+            INIT3S_led_on = true;
+            INIT3S_timer = timer_read();
+        }
+    }
+    if (INIT3S_led_on == true){
+        if(timer_elapsed(INIT3S_timer) > 2000){
+            eeconfig_init();
+            soft_reset_keyboard();
+        }
+    }
+
+    if (isRecording){
+        if(timer_elapsed(recording_timer) > 500)
+        {
+            isRecordingLedOn = !isRecordingLedOn;
+            recording_timer  = timer_read();
+        }
+    }
+}
+
+#ifdef DYNAMIC_MACRO_ENABLE
+void dynamic_macro_record_start_kb(int8_t direction) {
```

Comment:
hmm, this function doesn't appear to have a kb level callback ... which is an oversite.   For now: 
```suggestion
__attribute__((weak)) void dynamic_macro_record_start_user(int8_t direction) {
```
The weak attribute allows the user to replace the function without compiler errors, for now.

Reply:
I got it. thank you